src/Legacy/Storage.php

```diff
@@ -401,10 +401,10 @@ public function saveContent(Content $content, $comment = null)
         }
 
         // Test to see if this is a new record, or an update
-        if (empty($fieldvalues['id'])) {
-            $create = true;
-        } else {
-            $create = false;
+        $create = empty($fieldvalues['id']);
+
+        if (!isset($fieldvalues['slug'])) {
+            $fieldvalues['slug'] = ''; // Prevent 'slug may not be NULL'
```

Comment:
Oh, no you don't need to close either right now … this can can be adapted … the only thought is that not a lot should still use the legacy `saveContent()` … might be wroth hashing this over with Ross in Slack later